debugging/use-caution-when-reinitializing-variables-inside-a-loop.js

```diff
@@ -0,0 +1,21 @@
+function zeroArray(m, n) {
+  // Creates a 2-D array with m rows and n columns of zeroes
+  let newArray = [];
+  let row = [];
+
+  for (let i = 0; i < m; i++) {
+    // Adds the m-th row into newArray
+    row = [];
+
+  for (let j = 0; j < n; j++) {
```

Comment:
Here, we have two nested for loops.
The second for is inside the first one so you have to indent corrrectly. The inner loop must align with row = [] and newArray.push(row)